aws/resource_aws_xray_sampling_rule_test.go

```diff
@@ -0,0 +1,209 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSXraySamplingRuleConfig_update(ruleName, acctest.RandIntRange(0, 9999), acctest.RandIntRange(0, 2147483647), initialVersion),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckXraySamplingRuleExists(resourceName, &samplingRule),
+					testAccCheckResourceAttrRegionalARN(resourceName, "rule_arn", "xray", fmt.Sprintf("sampling-rule/%s", ruleName)),
+					resource.TestCheckResourceAttr(resourceName, "attributes.%", "0"),
+				),
+			},
+			{ // Update attributes
+				Config: testAccAWSXraySamplingRuleConfig_update(ruleName, updatedPriority, updatedReservoirSize, initialVersion),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckXraySamplingRuleExists(resourceName, &samplingRule),
+					testAccCheckResourceAttrRegionalARN(resourceName, "rule_arn", "xray", fmt.Sprintf("sampling-rule/%s", ruleName)),
+					resource.TestCheckResourceAttr(resourceName, "priority", fmt.Sprintf("%d", updatedPriority)),
+					resource.TestCheckResourceAttr(resourceName, "reservoir_size", fmt.Sprintf("%d", updatedReservoirSize)),
+					resource.TestCheckResourceAttr(resourceName, "attributes.%", "0"),
+				),
+			},
+			{ // Increment version
+				Config:      testAccAWSXraySamplingRuleConfig_update(ruleName, updatedPriority, updatedReservoirSize, updatedVersion),
+				ExpectError: regexp.MustCompile(`Version cannot be modified`),
+			},
+		},
+	})
+}
+
+func TestAccAWSXraySamplingRule_import(t *testing.T) {
+	resourceName := "aws_xray_sampling_rule.test"
+	rString := acctest.RandString(8)
+	ruleName := fmt.Sprintf("tf_acc_sampling_rule_%s", rString)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSIotThingTypeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSXraySamplingRuleConfig_basic(ruleName),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckXraySamplingRuleExists(n string, samplingRule *xray.SamplingRule) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No XRay Sampling Rule ID is set")
+		}
+
+		rule, err := getSamplingRule(rs.Primary.ID)
+
+		if err != nil {
+			return err
+		}
+
+		*samplingRule = *rule
+
+		return nil
+	}
+}
+
+func testAccCheckAWSXraySamplingRuleDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_xray_sampling_rule" {
+			continue
+		}
+
+		_, err := getSamplingRule(rs.Primary.ID)
+
+		if err == nil {
+			return fmt.Errorf("Expected XRay Sampling Rule to be destroyed, %s found", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+func getSamplingRule(ruleName string) (*xray.SamplingRule, error) {
```

Comment:
This function is super handy and should be used in the resource as well. I would recommend moving it to `aws/resource_aws_xray_sampling_rule.go` and updating the signature to `getXraySamplingRule(conn *xray.XRay, ruleName string) (*xray.SamplingRule, error)`